test_conformance/select/util_select.cpp

```diff
@@ -724,6 +769,30 @@ size_t check_long(void *test, void *correct, size_t count, size_t vector_size) {
     return 0;
 }
 
+size_t check_half(void *test, void *correct, size_t count, size_t vector_size)
+{
+    const cl_ushort *t = (const cl_ushort *)test;
```

Comment:
`*const t;`

You never want t to change either, const is massively underused in general (Not general here - general in the world!)

But more importantly why are they not

'const cl_half *const t;'

So that you can do log_error?

But if so log_error should have the cast - because it is the 'broken' bit - you have pointers to cl_halfs!